osquery/filesystem/darwin/plist.mm

```diff
@@ -112,21 +117,12 @@ Status filterArray(id plist, const std::string& root, pt::ptree& tree) {
       if (!status.ok()) {
         total_status = status;
       }
-    } else if ([value isKindOfClass:[NSData class]]) {
-      NSString* dataString = [value base64EncodedStringWithOptions:0];
-      child.put_value([dataString UTF8String]);
-    } else if ([value isKindOfClass:[NSDate class]]) {
-      NSNumber* seconds =
-          [[NSNumber alloc] initWithDouble:[value timeIntervalSince1970]];
-      child.put_value([[seconds stringValue] UTF8String]);
-    } else if ([value isEqual:@(YES)]) {
-      child.put_value("true");
-    } else if ([value isEqual:@(NO)]) {
-      child.put_value("false");
+    } else {
+      child.put_value(getValue(value));
     }
-    child_tree.push_back(std::make_pair("", child));
+    child_tree.push_back(std::make_pair("", std::move(child)));
   }
-  tree.push_back(pt::ptree::value_type(root, child_tree));
+  tree.push_back(ptvalue(root, std::move(child_tree)));
```

Comment:
I think that `ptvalue` should get nixed in favor of just using `pt::ptree::value_type` and letting clang-format take care of it, but that's obviously not related to this diff specifically.